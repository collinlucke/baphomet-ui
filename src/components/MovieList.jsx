import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

const Movie = ({ movie, deleteMovie }) => {
  const { title, year, rated, _id } = movie;
  const tdClassName = 'p-4 align-middle [&amp;:has([role=checkbox])]:pr-0';

  const deleteMovieHandler = () => {
    deleteMovie(_id);
  };

  return (
    <tr className="border-b transition-color hover:bg-muted/50 data-[state=selected:bg-muted">
      <td className={tdClassName}>{title}</td>
      <td className={tdClassName}>{year}</td>
      <td className={tdClassName}>{rated}</td>
      <td>
        <div className="flex gap-2">
          <Link
            to={`/edit/${_id}`}
            className="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-slate-100 h-9 rounded-md px-3"
          >
            Edit
          </Link>
          <button
            color="red"
            type="button"
            onClick={deleteMovieHandler}
            className="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-slate-100 hover:text-accent-foreground h-9 rounded-md px-3"
          >
            Delete
          </button>
        </div>
      </td>
    </tr>
  );
};

const MovieList = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    const getMovies = async () => {
      try {
        const response = await fetch(`http://localhost:5050/movies`);
        const movies = await response.json();
        setMovies(movies);
      } catch (err) {
        const message = `An error occurred: ${err.message}`;
        console.error(message);
        return;
      }
    };
    getMovies();
  }, [movies.length]);

  const deleteMovie = async id => {
    try {
      await fetch(`http://localhost:5050/movies/${id}`, {
        method: 'DELETE'
      });
      const newMovies = movies.filter(mov => {
        return mov._id !== id;
      });
      setMovies(newMovies);
    } catch (err) {
      const message = `An error occurred: ${err.message}`;
      console.error(message);
      return;
    }
  };

  const movieList = () => {
    return movies.map(mov => {
      return <Movie movie={mov} deleteMovie={deleteMovie} key={mov._id} />;
    });
  };

  return (
    <>
      <h3 className="text-lg font-semibold p-4">Movies</h3>
      <div className="border rounded-lg overflow-hidden">
        <div className="relative w-full overflow-auto">
          <table className="w-full caption-bottom text-sm">
            <thead className="[&amp;_tr]:border-b">
              <tr className="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted">
                <th className="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&amp;:has([role=checkbox])]:pr-0">
                  Title
                </th>
                <th className="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&amp;:has([role=checkbox])]:pr-0">
                  Year
                </th>
                <th className="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&amp;:has([role=checkbox])]:pr-0">
                  Rated
                </th>
                <th className="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&amp;:has([role=checkbox])]:pr-0">
                  Action
                </th>
              </tr>
            </thead>
            <tbody className="[&amp;_tr:last-child]:border-0">
              {movieList()}
            </tbody>
          </table>
        </div>
      </div>
    </>
  );
};

export default MovieList;
