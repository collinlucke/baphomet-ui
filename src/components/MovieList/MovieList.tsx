import { Button, List, Search } from '@collinlucke/phantomartist';
import { MovieListItem } from './MovieListItem';
import { ChangeEvent, FormEventHandler } from 'react';
import { useNavigate } from 'react-router-dom';
import { useScreenSize } from '../../hooks/useScreenSize';

type Movie = {
  id: string;
  title: string;
  releaseDate?: string;
  rated?: string;
  poster?: string;
};

type MovieData = {
  movieData: {
    allMovies: Movie[];
  };
  searchTerm?: string;
  resultsCount?: number;

  onSearch?: React.FormEventHandler<HTMLFormElement>;
  setSearchTerm?: (e: ChangeEvent<HTMLInputElement>) => void;
  openDeleteModal?: ({ id, title }: { id: string; title: string }) => void;
};

export const MovieList: React.FC<MovieData> = ({
  movieData,
  searchTerm,
  onSearch,
  setSearchTerm,
  openDeleteModal
}) => {
  const navigate = useNavigate();
  const screenSize = useScreenSize();

  const onSearchHandler: FormEventHandler<HTMLFormElement> = e => {
    onSearch?.(e);
  };

  const setSearchTermHandler = (e: ChangeEvent<HTMLInputElement>) => {
    setSearchTerm?.(e);
  };

  const navigateToArena = () => {
    navigate('/arena');
  };

  return (
    <List>
      <Search
        onSearch={onSearchHandler}
        searchTerm={searchTerm}
        searchLabel="Search Movies"
        setSearchTerm={setSearchTermHandler}
        resultsCount={movieData.allMovies.length}
        buttonSize={screenSize}
        inputSize={screenSize}
        useSearchButton={false}
      />
      {movieData.allMovies.length ? (
        movieData.allMovies.map(mov => (
          <MovieListItem
            mov={mov}
            key={mov.id}
            openDeleteModal={openDeleteModal || (() => {})}
          />
        ))
      ) : (
        <div css={[baphStyles.noResults]}>
          <h2>No Movies Match Your Search</h2>
          <div>Maybe you should go make one...</div>
        </div>
      )}
      <Button
        size={screenSize}
        onClick={navigateToArena}
        className={baphStyles}
      >
        Fight!
      </Button>
    </List>
  );
};

const baphStyles = {
  noResults: {
    display: 'flex',
    flexDirection: 'column' as 'column',
    alignItems: 'center',
    marginBottom: '30px',
    marginTop: '30px'
  },
  button: {
    alignSelf: 'end',
    display: 'flex',
    flexDirection: 'column' as 'column'
  }
};
