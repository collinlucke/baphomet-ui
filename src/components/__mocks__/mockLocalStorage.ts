export const mockLocalStorage = () => {
  localStorage.clear();
  const mockLocalStorage: Record<string, unknown> = {};
  global.localStorage = {
    length: 0,
    key: (index: number) => {
      const keys = Object.keys(mockLocalStorage);
      return keys[index] || null;
    },
    setItem: (key, value) => {
      mockLocalStorage[key] = value;
    },
    getItem: key => {
      const value = mockLocalStorage[key];
      return typeof value === 'string' ? value : null;
    },
    removeItem: key => {
      delete mockLocalStorage[key];
    },
    clear: () => {
      for (let key in mockLocalStorage) {
        delete mockLocalStorage[key];
      }
    }
  };
  return () => {};
};
