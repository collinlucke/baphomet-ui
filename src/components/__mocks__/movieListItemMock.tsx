export const createMovieListItemMock = () => ({
  MovieListItem: ({
    movie,
    openMovieDetails
  }: {
    movie: {
      id: string;
      title: string;
      winningPercentage: number;
      tmdbId: string;
    };
    openMovieDetails: (tmdbId: string) => void;
  }) => (
    <li data-testid={`movie-item-${movie.id}`}>
      <span>{movie.title}</span>
      <span data-testid={`winning-percentage-${movie.id}`}>
        {movie.winningPercentage}%
      </span>
      <button
        data-testid={`details-button-${movie.id}`}
        onClick={() => openMovieDetails(movie.tmdbId)}
      >
        Details
      </button>
    </li>
  )
});
