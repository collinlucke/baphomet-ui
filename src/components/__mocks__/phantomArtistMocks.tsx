import React from 'react';
import { vi } from 'vitest';

export const createPhantomArtistMock = () => ({
  List: vi.fn(
    ({
      children,
      ordered
    }: {
      children: React.ReactNode[];
      [key: string]: unknown;
    }) => {
      const ListElement = ordered ? 'ol' : 'ul';

      return <ListElement>{children}</ListElement>;
    }
  ),

  Search: vi.fn(
    ({
      searchTerm = '',
      setSearchTerm,
      searchLabel = 'Search',
      totalResultsCount = '0',
      showSearchButton = true,
      showResultsCount = true,
      buttonText = 'Search',
      label,
      autoFocus = false,
      testId = 'search-component',
      resultsAriaLive = 'polite'
    }: {
      searchTerm?: string;
      setSearchTerm?: (e: React.ChangeEvent<HTMLInputElement>) => void;
      searchLabel?: string;
      totalResultsCount?: string;
      showSearchButton?: boolean;
      showResultsCount?: boolean;
      buttonText?: string;
      label?: string;
      autoFocus?: boolean;
      testId?: string;
      resultsAriaLive?: 'polite' | 'assertive' | 'off';
    }) => {
      return (
        <div data-testid={testId} role="search" aria-label="Search">
          {showResultsCount && (
            <div
              data-testid="total-results"
              aria-live={resultsAriaLive}
              role="status"
            >
              Total Results: {totalResultsCount}
            </div>
          )}

          <input
            data-testid="search-input"
            type="search"
            value={searchTerm}
            onChange={setSearchTerm}
            placeholder={searchLabel}
            autoFocus={autoFocus}
            aria-label={label || searchLabel}
          />

          {showSearchButton && (
            <button
              data-testid="search-submit-button"
              type="button"
              aria-label={`${buttonText} - ${totalResultsCount} results available`}
            >
              {buttonText}
            </button>
          )}
        </div>
      );
    }
  ),

  Modal: ({
    isOpen,
    children,
    title,
    onClose
  }: {
    isOpen: boolean;
    children: React.ReactNode;
    title: string;
    onClose: () => void;
  }) =>
    isOpen ? (
      <div data-testid="modal" role="dialog">
        <h2>{title}</h2>
        <button onClick={onClose}>Close</button>
        {children}
      </div>
    ) : null,

  baseColors: {
    primary: {
      700: '#4A1512'
    },
    tertiary: { 50: '#ffffff' }
  },

  baseVibrantColors: {
    primary: {
      500: '#7B2D26',
      600: '#5A1E1A'
    },
    secondary: {
      500: '#2D5A87'
    },
    accent: {
      500: '#D4AF37'
    }
  },

  mediaQueries: {
    minWidth: {
      md: '@media (min-width: 768px)',
      lg: '@media (min-width: 1024px)',
      xl: '@media (min-width: 1280px)'
    }
  }
});
