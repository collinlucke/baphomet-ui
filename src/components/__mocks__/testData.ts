import { GET_MOVIE_BY_TMDB_ID } from '../../api/queries';

/**
 * Shared mock data for MovieList tests
 */
export const mockMovies = [
  {
    id: '1',
    title: 'The Matrix',
    releaseDate: '1999-03-31',
    posterUrl: 'https://example.com/matrix-poster.jpg',
    winningPercentage: 85.5,
    overview: 'A computer programmer discovers reality is a simulation.',
    genres: ['Action', 'Sci-Fi'],
    revenue: 463517383,
    backdropUrl: 'https://example.com/matrix-backdrop.jpg',
    tmdbId: '603'
  },
  {
    id: '2',
    title: 'The Godfather',
    releaseDate: '1972-03-24',
    posterUrl: 'https://example.com/godfather-poster.jpg',
    winningPercentage: 92.1,
    overview: 'The patriarch of an organized crime dynasty transfers control.',
    genres: ['Crime', 'Drama'],
    revenue: 246120974,
    backdropUrl: 'https://example.com/godfather-backdrop.jpg',
    tmdbId: '238'
  },
  {
    id: '3',
    title: 'Pulp Fiction',
    releaseDate: '1994-10-14',
    posterUrl: 'https://example.com/pulpfiction-poster.jpg',
    winningPercentage: 89.3,
    overview: 'The lives of two mob hitmen, a boxer, and others intertwine.',
    genres: ['Crime', 'Drama'],
    revenue: 214179088,
    backdropUrl: 'https://example.com/pulpfiction-backdrop.jpg',
    tmdbId: '680'
  }
];

export const mockMovieDetailsQuery = {
  request: {
    query: GET_MOVIE_BY_TMDB_ID,
    variables: { tmdbId: '603' }
  },
  result: {
    data: {
      movieResults: {
        id: '1',
        title: 'The Matrix',
        overview: 'A computer programmer discovers reality is a simulation.',
        releaseDate: '1999-03-31',
        posterUrl: 'https://example.com/matrix-poster.jpg',
        backdropUrl: 'https://example.com/matrix-backdrop.jpg',
        winningPercentage: 85.5,
        genres: ['Action', 'Sci-Fi'],
        revenue: 463517383,
        tmdbId: 603
      }
    }
  }
};

export const defaultMovieListProps = {
  movies: mockMovies,
  searchTerm: '',
  totalMovieCount: '3',
  showSearch: true,
  isLoadingMore: false,
  hasMore: true,
  onScroll: () => {},
  onSearch: () => {},
  setSearchTerm: () => {}
};
