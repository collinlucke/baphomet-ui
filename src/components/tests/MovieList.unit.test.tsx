import '@testing-library/jest-dom';
import { expect, describe, it, beforeEach, vi } from 'vitest';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MockedProvider } from '@apollo/client/testing/react';
import { createPhantomArtistMock } from '../__mocks__/phantomArtistMocks.tsx';
import { createMovieListItemMock } from '../__mocks__/movieListItemMock';
import { MovieList } from '../MovieList/MovieList';
import { setupIntersectionObserverMock } from '../__mocks__/browserMocks';
import { mockMovies, mockMovieDetailsQuery } from '../__mocks__/testData';

createPhantomArtistMock();
createMovieListItemMock();

describe('MovieList Component - Unit Tests', () => {
  const defaultProps = {
    movies: mockMovies,
    searchTerm: '',
    totalMovieCount: '3',
    isLoadingMore: false,
    hasMore: true,
    onScroll: vi.fn(),
    onSearch: vi.fn(),
    setSearchTerm: vi.fn()
  };

  beforeEach(() => {
    vi.clearAllMocks();

    setupIntersectionObserverMock();
  });

  describe('Component Logic & Props', () => {
    it('renders movie list when movies are provided', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      expect(screen.getByTestId('baph-movie-list')).toBeInTheDocument();
      expect(screen.getByTestId('movie-item-603')).toBeInTheDocument();
      expect(screen.getByTestId('movie-item-238')).toBeInTheDocument();
      expect(screen.getByTestId('movie-item-680')).toBeInTheDocument();
    });

    it('displays total movie count in search component', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} totalMovieCount="5" />
        </MockedProvider>
      );

      expect(screen.getByTestId('total-results')).toHaveTextContent('5');
    });

    it('handles empty and null movie arrays', () => {
      const { rerender } = render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} movies={[]} />
        </MockedProvider>
      );

      const noResultsLine1 = screen.getByText(
        'Real sorry to tell you this, but...'
      );
      const noResultsLine2 = screen.getByText('No movies match your search');

      expect(noResultsLine1).toBeInTheDocument();
      expect(noResultsLine2).toBeInTheDocument();

      rerender(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} movies={null} />
        </MockedProvider>
      );

      expect(noResultsLine1).toBeInTheDocument();
      expect(noResultsLine2).toBeInTheDocument();
    });
  });

  describe('Loading States', () => {
    it('shows loading indicator when isLoadingMore is true', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} isLoadingMore={true} />
        </MockedProvider>
      );

      expect(screen.getByText('Loading more movies...')).toBeInTheDocument();
    });

    it('shows end of results message when hasMore is false', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} hasMore={false} />
        </MockedProvider>
      );

      expect(
        screen.getByText("You've reached the end! ðŸŽ¬")
      ).toBeInTheDocument();
    });

    it('does not show end message when movies array is empty', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} movies={[]} hasMore={false} />
        </MockedProvider>
      );

      expect(
        screen.queryByText("You've reached the end! ðŸŽ¬")
      ).not.toBeInTheDocument();
    });
  });

  describe('Search Functionality', () => {
    it('calls setSearchTerm when input changes', async () => {
      const user = userEvent.setup();
      const mockSetSearchTerm = vi.fn();

      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} setSearchTerm={mockSetSearchTerm} />
        </MockedProvider>
      );

      const searchInput = screen.getByTestId('search-input');
      await user.type(searchInput, 'matrix');

      expect(mockSetSearchTerm).toHaveBeenCalled();
    });

    it('displays current search term', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} searchTerm="matrix" />
        </MockedProvider>
      );

      expect(screen.getByTestId('search-input')).toHaveValue('matrix');
    });
  });

  describe('Movie Details Functionality', () => {
    it('calls fetchMovieDetails when openMovieDetailsHandler is triggered', async () => {
      const user = userEvent.setup();

      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      const movieItem = screen.getByTestId('movie-item-603');
      await user.click(movieItem);

      await waitFor(() => {
        expect(movieItem).toBeInTheDocument();
      });
    });

    it('renders modal component', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
    });
  });

  describe('Scroll Functionality', () => {
    it('renders sentinel element for infinite scroll', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      const movieListWrapper =
        screen.getByTestId('baph-movie-list').parentElement;

      expect(movieListWrapper).toBeInTheDocument();
    });
  });
  describe('Accessibility', () => {
    it('has proper ARIA attributes', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      const movieList = screen.getByTestId('baph-movie-list');
      expect(movieList).toBeInTheDocument();

      const searchInput = screen.getByTestId('search-input');
      expect(searchInput).toHaveAttribute('placeholder', 'Search Movies');
    });

    it('renders movie items with proper structure', () => {
      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      const theMatrix = screen.getByTestId('movie-item-603');
      const theGodfather = screen.getByTestId('movie-item-238');
      const pulpFiction = screen.getByTestId('movie-item-680');
      expect(theMatrix).toBeInTheDocument();
      expect(theGodfather).toBeInTheDocument();
      expect(pulpFiction).toBeInTheDocument();

      expect(theMatrix).toHaveTextContent('85.5');
      expect(theGodfather).toHaveTextContent('92.1');
      expect(pulpFiction).toHaveTextContent('89.3');
    });
  });
  describe('Error Handling', () => {
    it('handles missing optional props gracefully', () => {
      const minimalProps = {
        movies: mockMovies,
        onScroll: vi.fn(),
        onSearch: vi.fn(),
        setSearchTerm: vi.fn()
      };

      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...minimalProps} />
        </MockedProvider>
      );

      expect(screen.getByTestId('baph-movie-list')).toBeInTheDocument();
      expect(screen.getByTestId('baph-search')).toBeInTheDocument();
    });

    it('handles undefined callback functions', () => {
      const propsWithUndefinedCallbacks = {
        ...defaultProps,
        onScroll: undefined,
        onSearch: undefined,
        setSearchTerm: undefined
      };

      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...propsWithUndefinedCallbacks} />
        </MockedProvider>
      );

      expect(screen.getByTestId('baph-movie-list')).toBeInTheDocument();
    });
  });

  describe('Custom CSS Classes', () => {
    it('applies custom className when provided', () => {
      const customClassName = {
        movieListWrapper: { backgroundColor: 'red' }
      };

      render(
        <MockedProvider mocks={[mockMovieDetailsQuery]}>
          <MovieList {...defaultProps} className={customClassName} />
        </MockedProvider>
      );

      expect(screen.getByTestId('baph-movie-list')).toBeInTheDocument();
    });
  });

  describe('Integration with Apollo Client', () => {
    it('integrates with Apollo MockedProvider without errors', () => {
      const mocks = [mockMovieDetailsQuery];

      render(
        <MockedProvider mocks={mocks}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      expect(screen.getByTestId('baph-movie-list')).toBeInTheDocument();
    });
  });
  describe('Snapshot Tests', () => {
    it('should match snapshot with default props', () => {
      const { container } = render(
        <MockedProvider mocks={[]}>
          <MovieList {...defaultProps} />
        </MockedProvider>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with empty movies list', () => {
      const { container } = render(
        <MockedProvider mocks={[]}>
          <MovieList {...defaultProps} movies={[]} />
        </MockedProvider>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with loading state', () => {
      const { container } = render(
        <MockedProvider mocks={[]}>
          <MovieList
            {...defaultProps}
            movies={[]}
            hasMore={true}
            isLoadingMore={true}
          />
        </MockedProvider>
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should match snapshot with loading state', () => {
    const { container } = render(
      <MockedProvider mocks={[]}>
        <MovieList
          {...defaultProps}
          movies={[]}
          hasMore={true}
          isLoadingMore={true}
        />
      </MockedProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with search functionality', () => {
    const { container } = render(
      <MockedProvider mocks={[]}>
        <MovieList
          {...defaultProps}
          onSearch={vi.fn()}
          searchTerm="test search"
        />
      </MockedProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
