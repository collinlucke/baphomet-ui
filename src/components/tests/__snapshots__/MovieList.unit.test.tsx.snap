// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieList Component - Unit Tests > Snapshot Tests > should match snapshot with default props 1`] = `
<div
  class="css-hhk6j-MovieList"
  data-testid="baph-movie-list"
>
  <div
    aria-label="Search"
    class="pa-search-form css-m5jsbo"
    data-testid="baph-search"
    role="search"
  >
    <div
      class="pa-search-wrapper css-114f50u"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="css-19gqcwz"
        data-testid="total-results"
        role="status"
      >
        Total Results:
         
        3
      </div>
      <div
        class="css-1jlnwea"
      >
        <div
          class="css-1mmmoc7"
        >
          <input
            aria-describedby="search-results-count"
            aria-invalid="false"
            aria-label="Search Movies"
            aria-required="false"
            class="css-jcn22a"
            data-testid="search-input"
            id="input-searchTerm"
            name="searchTerm"
            placeholder="Search Movies"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <ul
    class="pa-list css-179vcy7"
    role="list"
  >
    <li
      aria-label="Open details for The Matrix"
      class="css-zq0ncc"
      data-testid="movie-item-603"
      role="button"
      tabindex="0"
    >
      <div
        class="css-1u0mlrb"
      >
        <img
          alt="The Matrix"
          class="css-1u6rrjb"
          src="https://example.com/matrix-poster.jpg"
        />
        <div
          class="css-seut52"
        >
          <span
            class="css-2y6i2r"
          >
            85
            .
          </span>
          <span
            class="css-i0h9c4"
          >
            50
          </span>
        </div>
      </div>
    </li>
    <li
      aria-label="Open details for The Godfather"
      class="css-10225se"
      data-testid="movie-item-238"
      role="button"
      tabindex="0"
    >
      <div
        class="css-1u0mlrb"
      >
        <img
          alt="The Godfather"
          class="css-1u6rrjb"
          src="https://example.com/godfather-poster.jpg"
        />
        <div
          class="css-seut52"
        >
          <span
            class="css-2y6i2r"
          >
            92
            .
          </span>
          <span
            class="css-i0h9c4"
          >
            10
          </span>
        </div>
      </div>
    </li>
    <li
      aria-label="Open details for Pulp Fiction"
      class="css-1a9vumo"
      data-testid="movie-item-680"
      role="button"
      tabindex="0"
    >
      <div
        class="css-1u0mlrb"
      >
        <img
          alt="Pulp Fiction"
          class="css-1u6rrjb"
          src="https://example.com/pulpfiction-poster.jpg"
        />
        <div
          class="css-seut52"
        >
          <span
            class="css-2y6i2r"
          >
            89
            .
          </span>
          <span
            class="css-i0h9c4"
          >
            30
          </span>
        </div>
      </div>
    </li>
  </ul>
  <div
    class="css-hi4v9c"
  />
</div>
`;

exports[`MovieList Component - Unit Tests > Snapshot Tests > should match snapshot with empty movies list 1`] = `
<div
  class="css-hhk6j-MovieList"
  data-testid="baph-movie-list"
>
  <div
    aria-label="Search"
    class="pa-search-form css-m5jsbo"
    data-testid="baph-search"
    role="search"
  >
    <div
      class="pa-search-wrapper css-114f50u"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="css-19gqcwz"
        data-testid="total-results"
        role="status"
      >
        Total Results:
         
        3
      </div>
      <div
        class="css-1jlnwea"
      >
        <div
          class="css-1mmmoc7"
        >
          <input
            aria-describedby="search-results-count"
            aria-invalid="false"
            aria-label="Search Movies"
            aria-required="false"
            class="css-jcn22a"
            data-testid="search-input"
            id="input-searchTerm"
            name="searchTerm"
            placeholder="Search Movies"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-15vh70y-MovieList"
  >
    <h2>
      Real sorry to tell you this, but...
    </h2>
    <div>
      No movies match your search
    </div>
  </div>
  <div
    class="css-hi4v9c"
  />
</div>
`;

exports[`MovieList Component - Unit Tests > Snapshot Tests > should match snapshot with loading state 1`] = `
<div
  class="css-hhk6j-MovieList"
  data-testid="baph-movie-list"
>
  <div
    aria-label="Search"
    class="pa-search-form css-m5jsbo"
    data-testid="baph-search"
    role="search"
  >
    <div
      class="pa-search-wrapper css-114f50u"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="css-19gqcwz"
        data-testid="total-results"
        role="status"
      >
        Total Results:
         
        3
      </div>
      <div
        class="css-1jlnwea"
      >
        <div
          class="css-1mmmoc7"
        >
          <input
            aria-describedby="search-results-count"
            aria-invalid="false"
            aria-label="Search Movies"
            aria-required="false"
            class="css-jcn22a"
            data-testid="search-input"
            id="input-searchTerm"
            name="searchTerm"
            placeholder="Search Movies"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-15vh70y-MovieList"
  >
    <h2>
      Real sorry to tell you this, but...
    </h2>
    <div>
      No movies match your search
    </div>
  </div>
  <div
    class="css-hi4v9c"
  />
</div>
`;

exports[`MovieList Component - Unit Tests > should match snapshot with loading state 1`] = `
<div
  class="css-hhk6j-MovieList"
  data-testid="baph-movie-list"
>
  <div
    aria-label="Search"
    class="pa-search-form css-m5jsbo"
    data-testid="baph-search"
    role="search"
  >
    <div
      class="pa-search-wrapper css-114f50u"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="css-19gqcwz"
        data-testid="total-results"
        role="status"
      >
        Total Results:
         
        3
      </div>
      <div
        class="css-1jlnwea"
      >
        <div
          class="css-1mmmoc7"
        >
          <input
            aria-describedby="search-results-count"
            aria-invalid="false"
            aria-label="Search Movies"
            aria-required="false"
            class="css-jcn22a"
            data-testid="search-input"
            id="input-searchTerm"
            name="searchTerm"
            placeholder="Search Movies"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-15vh70y-MovieList"
  >
    <h2>
      Real sorry to tell you this, but...
    </h2>
    <div>
      No movies match your search
    </div>
  </div>
  <div
    class="css-hi4v9c"
  />
</div>
`;

exports[`MovieList Component - Unit Tests > should match snapshot with search functionality 1`] = `
<div
  class="css-hhk6j-MovieList"
  data-testid="baph-movie-list"
>
  <div
    aria-label="Search"
    class="pa-search-form css-m5jsbo"
    data-testid="baph-search"
    role="search"
  >
    <div
      class="pa-search-wrapper css-114f50u"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="css-19gqcwz"
        data-testid="total-results"
        role="status"
      >
        Total Results:
         
        3
      </div>
      <div
        class="css-1jlnwea"
      >
        <div
          class="css-1mmmoc7"
        >
          <input
            aria-describedby="search-results-count"
            aria-invalid="false"
            aria-label="Search Movies"
            aria-required="false"
            class="css-jcn22a"
            data-testid="search-input"
            id="input-searchTerm"
            name="searchTerm"
            placeholder="Search Movies"
            type="search"
            value="test search"
          />
        </div>
      </div>
    </div>
  </div>
  <ul
    class="pa-list css-179vcy7"
    role="list"
  >
    <li
      aria-label="Open details for The Matrix"
      class="css-zq0ncc"
      data-testid="movie-item-603"
      role="button"
      tabindex="0"
    >
      <div
        class="css-1u0mlrb"
      >
        <img
          alt="The Matrix"
          class="css-1u6rrjb"
          src="https://example.com/matrix-poster.jpg"
        />
        <div
          class="css-seut52"
        >
          <span
            class="css-2y6i2r"
          >
            85
            .
          </span>
          <span
            class="css-i0h9c4"
          >
            50
          </span>
        </div>
      </div>
    </li>
    <li
      aria-label="Open details for The Godfather"
      class="css-10225se"
      data-testid="movie-item-238"
      role="button"
      tabindex="0"
    >
      <div
        class="css-1u0mlrb"
      >
        <img
          alt="The Godfather"
          class="css-1u6rrjb"
          src="https://example.com/godfather-poster.jpg"
        />
        <div
          class="css-seut52"
        >
          <span
            class="css-2y6i2r"
          >
            92
            .
          </span>
          <span
            class="css-i0h9c4"
          >
            10
          </span>
        </div>
      </div>
    </li>
    <li
      aria-label="Open details for Pulp Fiction"
      class="css-1a9vumo"
      data-testid="movie-item-680"
      role="button"
      tabindex="0"
    >
      <div
        class="css-1u0mlrb"
      >
        <img
          alt="Pulp Fiction"
          class="css-1u6rrjb"
          src="https://example.com/pulpfiction-poster.jpg"
        />
        <div
          class="css-seut52"
        >
          <span
            class="css-2y6i2r"
          >
            89
            .
          </span>
          <span
            class="css-i0h9c4"
          >
            30
          </span>
        </div>
      </div>
    </li>
  </ul>
  <div
    class="css-hi4v9c"
  />
</div>
`;
